src/applications/disability-benefits/526EZ/containers/ITFWrapper.jsx

```diff
@@ -0,0 +1,174 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import LoadingIndicator from '@department-of-veterans-affairs/formation/LoadingIndicator';
+import AlertBox from '@department-of-veterans-affairs/formation/AlertBox';
+import AdditionalInfo from '@department-of-veterans-affairs/formation/AdditionalInfo';
+
+import { requestStates } from '../../../../platform/utilities/constants';
+import { itfStatuses } from '../constants';
+import { createITF as createITFAction, fetchITF as fetchITFAction } from '../actions';
+
+
+const fetchWaitingStates = [requestStates.notCalled, requestStates.pending];
+
+const noITFPages = ['/introduction', '/confirmation'];
+
+const itfMessage = (headline, content, status) => (
+  <div className="usa-grid" style={{ marginBottom: '2em' }}>
+    <AlertBox
+      isVisible
+      headline={headline}
+      content={content}
+      status={status}/>
+  </div>
+);
+
+const itfSuccessContent = (itf) => {
+  // TODO: Make this more better
+  // TODO: Format the date
+  const content = (
+    <div>
+      <p>Expiration date: {itf.currentITF.expirationDate}</p>
+      {itf.previousITF && <AdditionalInfo triggerText="Not what you expected?">
+        <p>
+          We found a previous ITF that expired on {itf.previousITF.expirationDate}. This might have been from an application that you started and did not finish before expiration, or from an earlier claim that you submitted.
+        </p>
+      </AdditionalInfo>}
+    </div>
+  );
+  return content;
+};
+
+
+export class ITFWrapper extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hasDisplayedSuccess: false
+    };
+  }
+
+
+  // When we first enter the form...
+  componentDidMount() {
+    // ...fetch the ITF if needed
+    if (!noITFPages.includes(this.props.location.pathname)) {
+      this.props.fetchITF();
+    }
+  }
+
+
+  componentWillReceiveProps(nextProps) {
+    const { itf, location } = nextProps;
+
+    if (noITFPages.includes(location.pathname)) {
+      return;
+    }
+
+    // We now know we've navigated to a page that requires ITF logic
+
+    if (itf.fetchCallState === requestStates.notCalled) {
+      nextProps.fetchITF();
+    }
+
+    // If we've already fetched the ITFs, have none active, and haven't already called createITF, submit a new ITF
+    const hasActiveITF = itf.currentITF && itf.currentITF.status === itfStatuses.active;
+    const createITFCalled = itf.creationCallState !== requestStates.notCalled;
+    if (itf.fetchCallState === requestStates.succeeded && !hasActiveITF && !createITFCalled) {
+      nextProps.createITF();
+    }
+
+    // If we've already displayed the itf success banner, toggle it off when the location changes
+    if (hasActiveITF && !this.state.hasDisplayedSuccess
```

Comment:
Hard to tell for sure, but does this set off any race conditions or other state wonkiness with regard to reliably displaying the success banner exactly one time?

Reply:
I haven't found any. Running through the lifecycle methods in my head, I couldn't figure any way to do break it.